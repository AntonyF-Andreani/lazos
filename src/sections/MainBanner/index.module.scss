.root {
  background-color: black;
  background-position: center;
  height: 100%;
  width: 100%;
  position: relative;
  justify-content: center;
  line-height: 0.5;

  &::before {
    content: " ";
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 2;
    background: #212529;
    opacity: 0.8;
    pointer-events: none;
    transition: ease opacity 300ms 100ms;
  }
}

.mainBanner {
  z-index: 1;
}

.image {
  width: 100%;
  height: 100vh;
  object-fit: cover;
  object-position: center;
}

.container {
  margin: 0 15px;
  width: auto;
  max-width: 750px;
  height: 100vh;
  justify-content: center;
  align-content: center;
  text-align: center;
  position: absolute;
  z-index: 3;
  margin-block: 15px;
}

.primaryText {
  text-align: center;
  color: #fff;
  font-weight: 700;
  font-size: 3.5em;
  line-height: 42px;
  letter-spacing: -0.05em;
  margin-bottom: 1rem;

  @media (max-width: 425px) {
    line-height: 1;
    font-size: 2em;
    margin-block: 5px;
  }

  @media (max-width: 640px), screen and (orientation: landscape) {
    line-height: 1;
    font-size: 2em;
    margin-block: 5px;
  }
}

.secondaryText {
  color: #fff;
  font-weight: 500;
  line-height: 1.5;
  font-size: 1.5em;
  padding: 0 1.5rem;
  display: inline-block;
  margin-bottom: 0.5rem;

  @media (max-width: 425px) {
    line-height: 1;
  }

  @media (max-width: 640px), screen and (orientation: landscape) {
    line-height: 1;
  }
}

.subtitleSlide {
  transform: translate3d(0, -50%, 0);
  opacity: 0.3;
}

.subtitle {
  min-height: 26px;
  color: #fff;
  font-weight: 300;
  opacity: 0.7;
  font-size: 1.2em;
  text-align: center;
  margin-bottom: 1.5rem;
  margin-top: 0.5rem;
  line-height: 26px;
  display: inline-block;

  @media (max-width: 425px) {
    margin-block: 15px;
    font-size: 1em;
  }

  @media (max-width: 640px), screen and (orientation: landscape) {
    margin-block: 15px;
    font-size: 1em;
  }
}

.button {
  margin: 0 15px 15px;

  @media (max-width: 425px) {
    padding: 1rem !important;
  }

  @media (max-width: 640px), screen and (orientation: landscape) {
    padding: 1rem !important;
  }
}
