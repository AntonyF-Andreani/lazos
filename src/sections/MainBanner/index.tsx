import React from "react";
import { Box, Grid, Typography } from "@mui/material";
import { LazyLoadImage } from "react-lazy-load-image-component";

import styles from "./index.module.scss";

import Button from "components/Button";
import type { Image } from "services/Cms/types";
export interface BannerProps {
  image: Image;
}

const MainBanner = ({ image }: BannerProps) => {
  return (
    <Grid container className={styles.root} id="inicio">
      <Box className={styles.mainBanner}>
        <LazyLoadImage
          alt={image.alt ? image.alt : "Background Image"}
          className={styles.image}
          effect="blur"
          src={image.filename}
        />
      </Box>
      <Grid container className={styles.container}>
        <Grid item>
          <Box>
            <LazyLoadImage
              alt={image.alt ? image.alt : "alt image"}
              className={styles.subtitleSlide}
              effect="blur"
              height="10px"
              src={"//a.storyblok.com/f/63950/37x10/a4291ce360/slide-title-border.png"}
              width="37px"
            />
            <Typography className={styles.secondaryText} variant="h3">
              CONECTANDO OPORTUNIDADES
            </Typography>
            <LazyLoadImage
              alt={image.alt ? image.alt : "alt image"}
              className={styles.subtitleSlide}
              effect="blur"
              height="10px"
              src={"//a.storyblok.com/f/63950/37x10/a4291ce360/slide-title-border.png"}
              width="37px"
            />
          </Box>
          <Typography className={styles.primaryText} variant="h2">
            Nueva Plataforma de Gestión Logística Social
          </Typography>
          <Typography className={styles.subtitle} variant="subtitle1">
            Presentanos tu solicitud de envío para llegar juntos a quienes más lo necesitan.
          </Typography>
        </Grid>
        <Grid item>
          <Button
            className={styles.button}
            href={
              "https://docs.google.com/forms/d/e/1FAIpQLSeHD6pa0cJ4AO8tgEWJsW3egaZ1VSOCPoioWW7g2Hfb2ZLr8Q/viewform"
            }
            text={"Quiero hacer mi primer traslado"}
            variant="contained"
          />
          <Button
            className={styles.button}
            href={
              "https://docs.google.com/forms/d/e/1FAIpQLSdyQtOoU6odC_hyt-88Z0m1YQPAKa3KofWo3blJvp_G41bRYA/viewform"
            }
            text={"Quiero volver a trasladar"}
            variant="contained"
          />
        </Grid>
      </Grid>
    </Grid>
  );
};

export default MainBanner;
