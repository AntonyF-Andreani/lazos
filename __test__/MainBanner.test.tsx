import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import MainBanner, { BannerProps } from "sections/MainBanner";

const defaultProps: BannerProps = {
  image: {
    filename: `banner`,
    alt: "test",
  },
};

describe("<MainBanner />", () => {
  it("should render without crash", () => {
    const { asFragment } = render(<MainBanner {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct image", () => {
    const { getAllByRole } = render(<MainBanner {...defaultProps} />);

    const images = getAllByRole("img");

    expect(images[0].getAttribute("src")).toBe(defaultProps.image.filename);
  });

  it("should have the correct alt of image", () => {
    const { getAllByRole } = render(<MainBanner {...defaultProps} />);

    const images = getAllByRole("img");

    expect(images[0].getAttribute("alt")).toBe(defaultProps.image.alt);
  });
});
