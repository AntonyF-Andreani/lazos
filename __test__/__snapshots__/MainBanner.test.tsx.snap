// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainBanner /> should render without crash 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container root css-11lq3yg-MuiGrid-root"
    id="inicio"
  >
    <div
      class="mainBanner MuiBox-root css-0"
    >
      <span
        class=" lazy-load-image-background blur"
        style="color: transparent; display: inline-block;"
      >
        <img
          alt="test"
          class="image"
          src="banner"
        />
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block; height: 10px; width: 37px;"
          >
            <img
              alt="test"
              class="subtitleSlide"
              height="10px"
              src="//a.storyblok.com/f/63950/37x10/a4291ce360/slide-title-border.png"
              width="37px"
            />
          </span>
          <h3
            class="MuiTypography-root MuiTypography-h3 secondaryText css-gepadz-MuiTypography-root"
          >
            CONECTANDO OPORTUNIDADES
          </h3>
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block; height: 10px; width: 37px;"
          >
            <img
              alt="test"
              class="subtitleSlide"
              height="10px"
              src="//a.storyblok.com/f/63950/37x10/a4291ce360/slide-title-border.png"
              width="37px"
            />
          </span>
        </div>
        <h2
          class="MuiTypography-root MuiTypography-h2 primaryText css-1sra7t5-MuiTypography-root"
        >
          Nueva Plataforma de Gestión Logística Social
        </h2>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 subtitle css-10wpov9-MuiTypography-root"
        >
          Presentanos tu solicitud de envío para llegar juntos a quienes más lo necesitan.
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <a
          class="MuiButtonBase-root MuiButton-root root button MuiButton-contained contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
          href="https://docs.google.com/forms/d/e/1FAIpQLSeHD6pa0cJ4AO8tgEWJsW3egaZ1VSOCPoioWW7g2Hfb2ZLr8Q/viewform"
          tabindex="0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Quiero hacer mi primer traslado
          </h6>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root root button MuiButton-contained contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
          href="https://docs.google.com/forms/d/e/1FAIpQLSdyQtOoU6odC_hyt-88Z0m1YQPAKa3KofWo3blJvp_G41bRYA/viewform"
          tabindex="0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Quiero volver a trasladar
          </h6>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
